src/olympia/devhub/tasks.py

```diff
@@ -187,9 +187,13 @@ def handle_upload_validation_result(results, upload_pk, channel):
     FileUpload instance."""
     upload = FileUpload.objects.get(pk=upload_pk)
 
-    if not upload.addon_id:
+    if (not upload.addon_id or
+            not upload.addon.find_latest_version(channel=channel, exclude=())):
```

Comment:
If a version has been deleted, it won't be found by `find_latest_version()`, so the new version will be considered the "first" one, restricting legacy. I think that's what we want here (avoid developers finding more weird workarounds bypassing the restriction)